src/lib.rs

```diff
@@ -287,39 +310,46 @@ mod tests {
 
     #[test]
     fn basic_expression_parse_test() {
-        let res = parser::ExprParser::new().parse("domain foo {}");
+        let mut errors = Vec::new();
+        let res = parser::ExprParser::new().parse(&mut errors, "domain foo {}");
```

Comment:
Not sure about that but it may be cleaner no move the `errors` argument from `parse()` to `new()`. It depends if `ExprParser` is usually stored in a variable and if multiple methods similar to `parse()` could be added. Making `ExprParser` owns `errors` could be an issue though.

Reply:
I don't *think* lalrpop exposes a way to do that.  This seems to be the recommended approach in their docs [here](http://lalrpop.github.io/lalrpop/tutorial/008_error_recovery.html) and [here](http://lalrpop.github.io/lalrpop/tutorial/009_state_parameter.html).